compact_str/src/repr/iter.rs

```diff
@@ -19,7 +19,7 @@ impl FromIterator<char> for Repr {
         // If the size hint indicates we can't store this inline, then create a heap string
         let (size_hint, _) = iter.size_hint();
         if size_hint > MAX_SIZE {
-            return Repr::from_string(iter.collect(), true);
+            return Repr::from_string(iter.collect(), true).unwrap();
```

Comment:
Is there anyway to avoid that string creation?

Reply:
Yes, and the code is much easier on the eyes, IMO.